presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -330,6 +330,7 @@ public PlanOptimizers(Metadata metadata, SqlParser sqlParser, FeaturesConfig fea
                 stats,
                 ImmutableList.of(new PartialAggregationPushDown(metadata.getFunctionRegistry())),
                 ImmutableSet.of(new PushPartialAggregationThroughJoin(), new PushPartialAggregationThroughExchange(metadata.getFunctionRegistry()))));
+        builder.add(new PruneUnreferencedOutputs());
```

Comment:
True. In any case. I don't think I can glue this iterative optimizer with the latter one since `AddIntermediateAggregations` might add intermediate aggregations before partial is pushed as deep as possible (making the final plan suboptimal)